### PR TITLE
add voter on node

### PR DESCRIPTION
**Pull request by veolys-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2173
https://github.com/open-orchestra/open-orchestra-front-bundle/pull/215
https://github.com/open-orchestra/open-orchestra-model-interface/pull/298
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/754
https://github.com/open-orchestra/open-orchestra-display-bundle/pull/255

### CODE REVIEW

File: FrontBundle/Controller/NodeController.php [hunk 1]
```diff
@@ -39,6 +41,15 @@ public function showAction(Request $request, $nodeId)
             throw new NonExistingNodeException();
         }
 
+        if (!empty($node->getFrontRoles())) {
+            if ($this->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_FULLY')) {
+                $this->denyAccessUnlessGranted(ContributionActionInterface::READ, $node);
+            } else {
+                throw new AuthenticationException();
+
```
Comment (alavieille) [inline FrontBundle/Controller/NodeController.php]: remove this line

**Comment by veolys-nanne** [comment]
update done @alavieille

**Comment by veolys-nanne** [comment]
update done @alavieille

File: FrontBundle/Controller/NodeController.php [hunk 2]
```diff
@@ -3,13 +3,15 @@
...
 use OpenOrchestra\FrontBundle\Exception\NonExistingNodeException;
+use OpenOrchestra\FrontBundle\Security\ContributionActionInterface;
 use OpenOrchestra\ModelInterface\Model\NodeInterface;
 use OpenOrchestra\ModelInterface\Model\ReadNodeInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration as Config;
 use OpenOrchestra\ModelInterface\Model\CacheableInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
```
Comment (alavieille) [inline FrontBundle/Controller/NodeController.php]: remove this use

File: FrontBundle/Security/Authorization/Voter/NodeVoter.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * Voter checking rights on node management
+ */
+class NodeVoter extends Voter implements ContainerAwareInterface
+{
+
+    private $container;
+    /**
+     * {@inheritDoc}
+     */
+    public function setContainer(ContainerInterface $container = null)
```
Comment (alavieille) [inline FrontBundle/Security/Authorization/Voter/NodeVoter.php]: why container aware, you can do like in RoleHierarchyVoter

File: FrontBundle/Tests/Security/Authorization/Voter/NodeVoter.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+use OpenOrchestra\ModelInterface\Model\ReadNodeInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+/**
+ * Class NodeVoter
+ *
+ * Voter checking rights on node management
+ */
+class NodeVoter extends Voter
```
Comment (alavieille) [inline FrontBundle/Tests/Security/Authorization/Voter/NodeVoter.php]: ??

**Pull request merged**